tools/ingest/rda/convert_rda_gdas.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+
+if (( ${#@} == 0 ))  ||  (( ${#@} > 2 )) || [[ $1 == [hH]elp ]]; then
+    echo "usage bash $0 yyyymmddhh [end_yyyymmddhh]"
+    echo "example:  bash $0 2017091500 [2017100112]"
+    exit 1
+elif (( ${#@} == 1 && ${#1} != 10 )); then
+    echo "unrecognized year-month-day: ${1}"
+    echo "usage bash $0 yyyymmddhh [end_yyyymmddhh]"
+    echo "example:  bash $0 2017091500 [2017100112]"
+    exit 1
+elif (( ${#@} == 2 && ${#1} != 10 )) || (( ${#@} == 2 && ${#2} != 10 )); then
+    echo "unrecognized year-month-day: ${1}  ${2}"
+    echo "usage bash $0 yyyymmddhh [end_yyyymmddhh]"
+    echo "example:  bash $0 2017091500 [2017100112]"
+    exit 1
+fi
+
+dtg=${1}
+final_dtg=${2:-${dtg}}
+
+assim_freq=${assim_freq:-6}
+window_length=$( printf "%d" ${assim_freq} )
+
+workdir=${UWORKDIR:-/glade/p/univ/ucos0003/${USER}}
+data_type=${data_type:-amsua}
+data_gdas_type=${data_gdas_type:-1bamua}
+data_repos=${DATA_REPOS:-${workdir}/data_repos}
+ioda_bundle=${IODA_BUNDLE:-${WORKDIR}/work/JCSDA/JEDI/ioda-bundle}
+ncar2ioda_bundle=${NCAR2IODA_BUNDLE:-${WORKDIR}/work/JCSDA/JEDI/ncar2ioda/obs2ioda/obs2ioda-v2/src}
+# can use for all ?
+# ncar2ioda_bundle=${NCAR2IODA_BUNDLE:-${WORKDIR}/work/JCSDA/JEDI/ncar2ioda/obs2ioda/obs2ioda-prepBUFR/src}
+
+
+gen_ioda() {
+    cd ${output_temp_dir}
+    wild_string=${data_type}_*_obs_${dtg}.h5
+    nfiles=$(ls -1 ${wild_string}  2>/dev/null | wc -l)
+    if (( nfiles == 0 )); then
+        echo " ... WARNING no files found: ${output_temp_dir}/${wild_string}"
+        return 1
+    fi
+    for afile in ${wild_string}; do
+        platform=${afile#${data_type}_}
+        platform=${platform%_obs*}
+        echo "   .... ahhh ... platform ... ${platform}"
+        output_file="${output_dir}/${atime}_${data_type}_${platform}.nc4"
+        cmd="mpirun -np 1 ${ioda_bundle}/build/bin/ioda-upgrade.x ${afile} ${output_file}"
+        echo "${cmd}"
+        ${cmd}
+        # path to utility does not work
+        python3 ${UWORKDIR}/data_repos/fix_nchans.py --input ${output_file} --sensor ${data_type}
```

Comment:
this also only needs to be called for radiances `amsua`, `mhs`, `iasi`, `cris` but does not need to be called for `prepBUFR` for instance